cmd/agent/app/run.go

```diff
@@ -282,9 +283,12 @@ func StartAgent() error {
 		http.Handle("/telemetry", telemetryHandler)
```

Comment:
This handler must now be added to `common.ExpvarServer` (so `common.ExpvarServer` needs to be instantiated before this line of code).

Handlers that are added with `http.Handle()` are added to the default HTTP server, so they must now be added to `common.ExpvarServer` instead.

(I haven't checked the rest of the code, but if `http.Handle()` is called directly in other places of the core agent code, similarly these handlers must now be added to `common.ExpvarServer` instead)

Reply:
@olivielpeau 
Thank you for your review.

> Handlers that are added with http.Handle() are added to the default HTTP server, so they must now be added to common.ExpvarServer instead.

I'm not certain. Do you mean that the code should be `common.ExpvarServer.Handle("/telemetry", telemetryHandler)`?
`common.ExpvarServer` is a [http.Server](https://pkg.go.dev/net/http#Server) type, so it doesn't have `Handle()` method.

I'm sorry for my lack of understanding, but could you please give me a sample of the code?

> (I haven't checked the rest of the code, but if http.Handle() is called directly in other places of the core agent code, similarly these handlers must now be added to common.ExpvarServer instead)

In [StartAgent()](https://github.com/keisku/datadog-agent/blob/f6e82d68b5d928ac163c42bd793c2a9233216e39/cmd/agent/app/run.go#L162-L470) method, [L283](https://github.com/keisku/datadog-agent/blob/f6e82d68b5d928ac163c42bd793c2a9233216e39/cmd/agent/app/run.go#L283) is the only line to use `http.Handle()`

Reply:
My bad! I missed that you're using `http.DefaultServeMux` as the handler of `common.ExpvarServer`, so indeed the `http.Handle` call here still works.

So please ignore this comment, sorry for the confusion.